cpp/include/experimental/graph.hpp

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <experimental/graph_view.hpp>
+#include <utilities/error.hpp>
+
+#include <raft/handle.hpp>
+#include <rmm/device_uvector.hpp>
+
+#include <cstddef>
+#include <string>
+#include <type_traits>
+#include <vector>
+
+namespace cugraph {
+namespace experimental {
+
+template <typename vertex_t, typename edge_t, typename weight_t>
+struct edgelist_t {
+  vertex_t const *p_src_vertices{nullptr};
+  vertex_t const *p_dst_vertices{nullptr};
+  weight_t const *p_edge_weights{nullptr};
+  edge_t number_of_edges{0};
+};
+
+template <typename vertex_t,
```

Comment:
Not sure the best place to put it, but it would be useful to document the difference between graph_t and graph_view_t somewhere in the definition of the classes.

Reply:
Added

// graph_t is an owning graph class (note that graph_view_t is a non-owning graph class)

and 

https://github.com/rapidsai/cugraph/pull/1098/files#diff-75b979f478be47f71bd6933f474074cbR128

as well